packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/method_channel/method_channel_firebase_remote_config.dart

```diff
@@ -0,0 +1,294 @@
+  ///
+  /// When the user code calls a Remote Config method, the real instance
+  /// is initialized via the [delegateFor] method.
+  MethodChannelFirebaseRemoteConfig._() : super(appInstance: null);
+
+  /// Creates a new instance for a given [FirebaseApp].
+  MethodChannelFirebaseRemoteConfig({@required FirebaseApp app})
+      : super(appInstance: app);
+
+  Map<String, RemoteConfigValue> _activeParameters;
+  RemoteConfigSettings _settings;
+  DateTime _lastFetchTime;
+  RemoteConfigFetchStatus _lastFetchStatus;
+
+  /// Gets a [FirebaseRemoteConfigPlatform] instance for a specific
+  /// [FirebaseApp].
+  ///
+  /// Instances are cached and reused for incoming event handlers.
+  @override
+  FirebaseRemoteConfigPlatform delegateFor({FirebaseApp app}) {
+    if (_methodChannelFirebaseRemoteConfigInstances.containsKey(app.name)) {
+      return _methodChannelFirebaseRemoteConfigInstances[app.name];
+    }
+
+    _methodChannelFirebaseRemoteConfigInstances[app.name] =
+        MethodChannelFirebaseRemoteConfig(app: app);
+    return _methodChannelFirebaseRemoteConfigInstances[app.name];
+  }
+
+  @override
+  FirebaseRemoteConfigPlatform setInitialValues(
+      {Map<dynamic, dynamic> remoteConfigValues}) {
+    final fetchTimeout = Duration(seconds: remoteConfigValues['fetchTimeout']);
+    final minimumFetchInterval =
+        Duration(seconds: remoteConfigValues['minimumFetchInterval']);
+    final lastFetchMillis = remoteConfigValues['lastFetchTime'];
+    final lastFetchStatus = remoteConfigValues['lastFetchStatus'];
+
+    _settings = RemoteConfigSettings(
+      fetchTimeout: fetchTimeout,
+      minimumFetchInterval: minimumFetchInterval,
+    );
+    _lastFetchTime = DateTime.fromMillisecondsSinceEpoch(lastFetchMillis);
+    _lastFetchStatus = _parseFetchStatus(lastFetchStatus);
+    _activeParameters = _parseParameters(remoteConfigValues['parameters']);
+    return this;
+  }
+
+  RemoteConfigFetchStatus _parseFetchStatus(String status) {
+    switch (status) {
+      case 'noFetchYet':
+        return RemoteConfigFetchStatus.noFetchYet;
+      case 'success':
+        return RemoteConfigFetchStatus.success;
+      case 'failure':
+        return RemoteConfigFetchStatus.failure;
+      case 'throttle':
+        return RemoteConfigFetchStatus.throttle;
+      default:
+        return RemoteConfigFetchStatus.noFetchYet;
+    }
+  }
+
+  @override
+  DateTime get lastFetchTime => _lastFetchTime;
+
+  @override
+  RemoteConfigFetchStatus get lastFetchStatus => _lastFetchStatus;
+
+  @override
+  RemoteConfigSettings get settings => _settings;
+
+  @override
+  set settings(RemoteConfigSettings remoteConfigSettings) {
+    _settings = remoteConfigSettings;
+  }
+
+  @override
+  Future<void> ensureInitialized() async {
+    try {
+      await channel.invokeMethod<void>(
+          'RemoteConfig#ensureInitialized', <String, dynamic>{
+        'appName': app.name,
+      });
+    } catch (e) {
+      throw convertPlatformException(e);
```

Comment:
Missing the stack trace being passed down here (recent-ish addition to core). The second arg of  `convertPlatformException` should pass an optional `StackTrace`.

Reply:
Would `StackTrace.current` be sufficient here?

Reply:
The catch provides it:

```
} catch (e, s) {
  throw convertPlatformException(e, s);
```

Reply:
Ah! Thanks: done.